src/widgets/BreachCard.js

```diff
@@ -0,0 +1,78 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import { Text, StyleSheet } from 'react-native';
+import { DARKER_GREY, APP_FONT_FAMILY, COLD_BREACH_BLUE, DANGER_RED } from '../globalStyles';
+
+import { ColdBreachIcon, HotBreachIcon, Paper } from '.';
+import { NoBreachMan } from './NoBreachMan';
+import { vaccineStrings } from '../localization/index';
+
+export const BreachCard = ({ type, breachCount, headerText, message }) => {
+  let icon = null;
+  let textStyle = {};
+  let text = breachCount?.toString();
+
+  switch (type) {
+    case 'cold':
+      icon = <ColdBreachIcon color={COLD_BREACH_BLUE} />;
+      textStyle = localStyles.coldText;
+      break;
+    case 'hot':
+      icon = <HotBreachIcon color={DANGER_RED} />;
+      textStyle = localStyles.hotText;
+      break;
+    case 'text':
+      textStyle = [localStyles.hotText, { color: DARKER_GREY }];
+      text = message;
+      break;
+    default:
+      break;
+  }
+
+  if (!breachCount && type !== 'text') {
+    icon = <NoBreachMan />;
+    text = vaccineStrings.no_breaches;
+    textStyle = localStyles.noBreachText;
+  }
+
+  return (
+    <Paper
+      headerText={headerText}
+      style={localStyles.card}
+      contentContainerStyle={{ alignItems: 'center', paddingBottom: 10 }}
+    >
+      <Text style={textStyle}>{text}</Text>
+      {icon}
+    </Paper>
+  );
+};
+
```

Comment:
I like this a lot, thanks.

I was wondering if we should actually take it further: the [mapStateToProps](https://github.com/openmsupply/mobile/blob/55ebde82acaf95465f393eb9859ed449db053c10/src/pages/FridgeDetailPage.js#L226-L276) is a bit of an issue as part of #3386 I was going to do this also- but was thinking to `connect` each breach card to redux directly, which would spread the time taken for the map state to props function and not front load it.

Reply:
Nice! I've had a go, with this: c812b886

Wasn't sure how else to keep the shared elements in a single component while only referencing a specific part of the state. I think that this is quicker to load..?